crypto/s2n_ecc_evp.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "crypto/s2n_ecc_evp.h"
+
+#include <openssl/ecdh.h>
+#include <openssl/evp.h>
+#include <stdint.h>
+
+#include "tls/s2n_tls_parameters.h"
+#include "utils/s2n_mem.h"
+#include "utils/s2n_safety.h"
+
+DEFINE_POINTER_CLEANUP_FUNC(EVP_PKEY *, EVP_PKEY_free);
+DEFINE_POINTER_CLEANUP_FUNC(EVP_PKEY_CTX *, EVP_PKEY_CTX_free);
+
+/* IANA values can be found here:
+ * https://tools.ietf.org/html/rfc8446#appendix-B.3.1.4 */
+/* Share sizes are described here:
+ * https://tools.ietf.org/html/rfc8446#section-4.2.8.2 and include the extra
+ * "legacy_form" byte */
+#if S2N_OPENSSL_VERSION_AT_LEAST(1, 1, 0) && !defined(LIBRESSL_VERSION_NUMBER)
```

Comment:
perhaps define `S2N_OPENSSL_VERSION_AT_LEAST(1, 1, 0) && !defined(LIBRESSL_VERSION_NUMBER)` as `S2N_IS_X25519_SUPPORTED` so that it could be reused for the other conditionals?

Reply:
good catch, will add